src/traceViewer.ts

```diff
@@ -18,21 +18,37 @@ import { ChildProcess, spawn } from 'child_process';
 import type { TestConfig } from './playwrightTestTypes';
 import { findNode } from './utils';
 import * as vscodeTypes from './vscodeTypes';
+import { SettingsModel } from './settingsModel';
 
-export type TraceViewer = SpawnTraceViewer;
+export type TraceViewer = {
+  isStarted(): boolean;
+  currentFile(): string | undefined;
+  willRunTests(): Promise<void>;
+  open(file: string): Promise<void>;
+  close(): void;
+  checkSupport(userGesture?: boolean): boolean;
+  infoForTest(): {
+    type: string;
+    serverUrlPrefix: string;
+    testConfigFile: string;
+    traceFile: string | undefined;
+  } | undefined;
+};
 
-export class SpawnTraceViewer {
+export class SpawnTraceViewer implements TraceViewer {
   private _vscode: vscodeTypes.VSCode;
+  private _settingsModel: SettingsModel;
   private _envProvider: () => NodeJS.ProcessEnv;
   private _traceViewerProcess: ChildProcess | undefined;
   private _currentFile?: string;
   private _config: TestConfig;
   private _serverUrlPrefixForTest?: string;
 
-  constructor(vscode: vscodeTypes.VSCode, envProvider: () => NodeJS.ProcessEnv, config: TestConfig) {
+  constructor(vscode: vscodeTypes.VSCode, settingsModel: SettingsModel, envProvider: () => NodeJS.ProcessEnv, config: TestConfig) {
```

Comment:
nit: Passing settingsModel does not seem justified atm, but I'm fine with it

Reply:
I think changes I'll do on https://github.com/microsoft/playwright-vscode/pull/513#discussion_r1706132388 will no longer require this dependency on `settingsModel`

Reply:
Addressed in 31a15f3f926c5f17a8e16a4476d52c62f11778a9